dbt_meshify/dbt_projects.py

```diff
@@ -242,13 +261,51 @@ class DbtSubProject(BaseDbtProject):
     def __init__(self, name: str, parent_project: DbtProject, resources: Set[str]):
         self.name = name
         self.resources = resources
-        self.parent = parent_project
+        self.parent_project = parent_project
 
-        self.manifest = parent_project.manifest.deepcopy()
+        # self.manifest = parent_project.manifest.deepcopy()
+        # i am running into a bug with the core deepcopy -- checking with michelle
+        self.manifest = copy.deepcopy(parent_project.manifest)
         self.project = copy.deepcopy(parent_project.project)
         self.catalog = parent_project.catalog
+        self.custom_macros = self._get_custom_macros()
+        self.file_manager = DbtFileManager(
+            read_project_path=parent_project.path, write_project_path=Path(name)
+        )
 
-        super().__init__(self.manifest, self.project, self.catalog)
+        self._rename_project()
+
+        super().__init__(self.manifest, self.project, self.catalog, self.name)
+
+    def _rename_project(self) -> None:
+        """
+        edits the project yml to take any instance of the parent project name and update it to the subproject name
+        """
+        project_dict = self.project.to_dict()
+        for key in [resource.pluralize() for resource in NodeType]:
+            if self.parent_project.name in project_dict.get(key, {}).keys():
+                project_dict[key][self.name] = project_dict[key].pop(self.parent_project.name)
+        project_dict["name"] = self.name
+        self.project = Project.from_dict(project_dict)
+
+    def _get_custom_macros(self) -> Set[str]:
+        """
+        get a set of macro unique_ids for all the selected resources
+        """
+        macro_list = []
```

Comment:
Since we're converting `macro_list` to a `Set` on line 308, it may make sense to initialize this as a set from the beginning. This will reduce the memory usage of `project_macros` since we're not storing duplicate values in `List`.